### PR TITLE
3328 npe grideditor

### LINKED ISSUE #3328: NPE While clicking inside Grid Editor view after plot is closed
### Steps to Reproduce

1. Open nosuch.rep file from sr2 folder
2. drag sensor.dsf file
3. Open grid editor view
4. Select 826 positions from nosuch track layer
5. Now Grid Editor View populates track data
6. Close nosuch.rep editor
7. See that the contents in grid editor view are still exist/displayed - 
8. Click inside the Grid editor view - see that NPE error dialog is displayed - https://gyazo.com/d08dc420a70a19339237865cbe5dfced , with the following stack trace 
[NPE_Grid_Editor_view.txt](https://github.com/debrief/debrief/files/2502064/NPE_Grid_Editor_view.txt)

### Expected Result

- While closing the plot, the data inside the grid editor view should be removed - grid view should be empty
- No Exception should be logged while selecting inside this view.

### PR DESCRIPTION
**Pull request by ayeshare**
Fixes #3328

### CODE REVIEW

File: org.mwc.cmap.NarrativeViewer/src/org/mwc/cmap/NarrativeViewer2/FilteredNatTable.java [hunk 1]
```diff
@@ -239,6 +239,12 @@ protected void clearText()
     textChanged();
   }
 
+  private Image getImageFromJFaceResources(String imageDes) {
+    if(JFaceResources.getImageRegistry().getDescriptor(imageDes)!=null) {
```
[the remark this answers is not in the document]
Comment (ayeshare) [reply org.mwc.cmap.NarrativeViewer/src/org/mwc/cmap/NarrativeViewer2/FilteredNatTable.java]: done

File: org.mwc.cmap.grideditor/src/org/mwc/cmap/grideditor/GridEditorView.java [hunk 2]
```diff
@@ -66,13 +76,55 @@ public void createPartControl(final Composite parent)
...
+		_myPartMonitor.addPartListener(CorePlotEditor.class, PartMonitor.CLOSED, new PartMonitor.ICallback()
+    {
+      
+      @Override
+      public void eventTriggered(String type, Object instance,
+          IWorkbenchPart parentPart)
+      {
+        if(instance.equals(displayedPlot)) {
+          //set input null now.
+          activePlots.remove(instance);
```
Comment (IanMayo) [inline org.mwc.cmap.grideditor/src/org/mwc/cmap/grideditor/GridEditorView.java]: @ayeshare - we never use the contents of activePlots.  I think we don't need them. Could you check?

**Comment by IanMayo** [comment]
@ayeshare - it was a tricky issue you had to resolve here. I hadn't realised that it was the `Selection` that we were using to control what was displayed, not the part provider.  I've highlighted a object that appears unused, but other than that it's a good solution, thanks :-)

**Comment by ayeshare** [comment]
> 
> 
> @ayeshare - it was a tricky issue you had to resolve here. I hadn't realised that it was the `Selection` that we were using to control what was displayed, not the part provider. I've highlighted a object that appears unused, but other than that it's a good solution, thanks :-)

Which object appears unused? I couldnt understand from your comment

**Comment by IanMayo** [comment]
IIRC, it as `activePlots` in GridEditorView that never got read.

**Pull request merged**